scripts/keybindings/6_vscode_integration.cjs

```diff
@@ -1,98 +1,68 @@
-    },
-    {
-        key: "ctrl+w =",
-        command: "workbench.action.evenEditorWidths",
-        when: "!editorTextFocus && !terminalFocus",
-    },
-    {
-        key: "ctrl+w _",
-        command: "workbench.action.toggleEditorWidths",
-        when: "!editorTextFocus && !terminalFocus",
-    },
-    {
-        key: "ctrl+w >",
-        command: "workbench.action.increaseViewWidth",
-        when: "!editorTextFocus && !terminalFocus",
-    },
-    {
-        key: "ctrl+w <",
-        command: "workbench.action.decreaseViewWidth",
-        when: "!editorTextFocus && !terminalFocus",
-    },
-    {
-        key: "ctrl+w +",
-        command: "workbench.action.increaseViewHeight",
-        when: "!editorTextFocus && !terminalFocus",
-    },
-    {
-        key: "ctrl+w -",
-        command: "workbench.action.decreaseViewHeight",
-        when: "!editorTextFocus && !terminalFocus",
-    },
-    {
-        key: "ctrl+w s",
-        command: "workbench.action.splitEditorDown",
-        when: "!editorTextFocus && !terminalFocus",
-    },
-    {
-        key: "ctrl+w v",
-        command: "workbench.action.splitEditorRight",
-        when: "!editorTextFocus && !terminalFocus",
-    },
-];
+const { addKeybinds } = require("./util.cjs");
+
+const [add, keybinds] = addKeybinds();
+
+add("ctrl+w", null, null, "-workbench.action.switchWindow");
+add(
+    "ctrl+w ctrl+w",
+    "!editorTextFocus && !terminalFocus && !(filesExplorerFocus || inSearchEditor || searchViewletFocus || replaceInputBoxFocus)",
+    null,
+    "workbench.action.focusNextGroup",
+);
+for (const [key, cmd] of [
+    ["ctrl+w up", "workbench.action.navigateUp"],
+    ["ctrl+w k", "workbench.action.navigateUp"],
+    ["ctrl+w down", "workbench.action.navigateDown"],
+    ["ctrl+w j", "workbench.action.navigateDown"],
+    ["ctrl+w left", "workbench.action.navigateLeft"],
+    ["ctrl+w h", "workbench.action.navigateLeft"],
+    ["ctrl+w right", "workbench.action.navigateRight"],
+    ["ctrl+w l", "workbench.action.navigateRight"],
+    ["ctrl+w =", "workbench.action.evenEditorWidths"],
+    ["ctrl+w _", "workbench.action.toggleEditorWidths"],
+    ["ctrl+w >", "workbench.action.increaseViewWidth"],
+    ["ctrl+w <", "workbench.action.decreaseViewWidth"],
+    ["ctrl+w +", "workbench.action.increaseViewHeight"],
+    ["ctrl+w -", "workbench.action.decreaseViewHeight"],
+    ["ctrl+w s", "workbench.action.splitEditorDown"],
+    ["ctrl+w v", "workbench.action.splitEditorRight"],
+]) {
+    add(key, "!editorTextFocus && !terminalFocus", null, cmd);
+}
+
+// - Why do we need to manually send keys?
+//    The output panel is not a "real" text editor, it cannot receive regular keyboard inputs.
+// - Why do we send these keys?
+//    We send certain basic keys for actions like moving, selecting, and copying.
+//    This allows us to easily view the output content and copy it, such as error messages.
+for (const item of [
+    // motion
+    "h",
```

Comment:
may as well send ALL `type` events (A-Z, 0-9, symbols, bspc/del/left/right/up/down/tab)

Reply:
The output is read-only

Reply:
Something like `yip` might still want to be supported though. Is there a disadvantage to supporting all keys?

Reply:
Or `v]ip"+y`.

Reply:
Need to avoid any key that may modify buffer content and enter insert mode. Because the output is read-only in vscode, changes made in nvim cannot be applied to vscode.

Reply:
I thought the output buffer was readonly in vim as well.

Reply:
This use case is not very common. Basic movement and copying are already sufficient

Reply:
Fair enough. I just want to reduce edge cases/inconsistency when possible.

Reply:
And things like support custom textobjects for copying. (Maybe someone has a url textobj they frequently use with `u`).

Reply:
This use case itself is an "edge case". I also want to be able to quickly copy using the textobject, but considering if more keys are added, it may lead to more "non-value" problems due to user-defined mappings in nvim.

Reply:
Again, can't we just set the output to readonly in nvim?

Reply:
Cannot. Because we need to apply vscode changes to nvim

Reply:
But I have thought of a solution, which is to temporarily modify the readonly and modifiable options of the buffer when applying vscode changes.

Reply:
I was about to suggest that!